adbms/src/test/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelationTest.scala

```diff
@@ -0,0 +1,270 @@
+package de.up.hpi.informationsystems.adbms.definition
+
+import org.scalatest.{AsyncWordSpec, Matchers, WordSpec}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.concurrent.Future
+import scala.util.Success
+
+class FutureRelationTest extends WordSpec with Matchers {
+
+  "A future relation" when {
+
+    val colFirstname: ColumnDef[String] = ColumnDef("Firstname")
+    val colLastname: ColumnDef[String] = ColumnDef("Lastname")
+    val colAge: ColumnDef[Int] = ColumnDef("Age")
+
+    val columns: Set[UntypedColumnDef] = Set(colFirstname, colLastname, colAge)
+
+    val record1 = Record(columns)
+      .withCellContent(colFirstname)("Test")
+      .withCellContent(colLastname)("Test")
+      .withCellContent(colAge)(42)
+      .build()
+
+    val record2 = Record(Set(colFirstname, colLastname, colAge))
+      .withCellContent(colFirstname)("Max")
+      .withCellContent(colLastname)("Mustermann")
+      .withCellContent(colAge)(23)
+      .build()
+
+    val record3 = Record(columns)
+      // missing firstName
+      .withCellContent(colLastname)("es")
+      .withCellContent(colAge)(200215)
+      .build()
+
+    val record4 = Record(columns)
+      .withCellContent(colFirstname)(null)
+      .withCellContent(colAge)(2)
+      .build()
+
+    "empty" should {
+      val futureEmpty: Future[Seq[Record]] = Future {
+        Thread.sleep(200)
+        Seq.empty
+      }
+      val emptyRelation = FutureRelation.fromRecordSeq(futureEmpty)
+
+      "return an empty result set for any where or whereAll query" in {
```

Comment:
we already checked this behavior in `TransientRelation`, so you don't have to copy it here for our small project ;)
But as you already did, we keep it there.

Reply:
You're right in that this is a little redundant. I wanted to make sure that it still works wrapped in a Future.